src/org/labkey/test/tests/component/GridPanelViewTest.java

```diff
@@ -0,0 +1,1389 @@
+        expectedValues.add(viewName1);
+        actualValues = manageViewsDialog.getViewNames();
+        checker().withScreenshot("Manage_Views_Saved_Other_User_Error")
+                .verifyEqualsSorted(String.format("List of view in 'Manage Views' dialog not as expected for user '%s' after saving view '%s'.", OTHER_USER, viewName1),
+                        expectedValues, actualValues);
+
+        manageViewsDialog.dismiss();
+
+        stopImpersonating();
+
+    }
+
+    /**
+     * Helper to validate the 'Views' menu.
+     *
+     * @param screenShotPrefix A string to put at the start of the screenshot name (if taken).
+     * @param grid A reference to the QueryGrid.
+     * @param expectedMenuItems A list of the expected menu items.
+     * @return True if no errors, false otherwise.
+     */
+    private boolean validateViewMenu(String screenShotPrefix, QueryGrid grid, List<String> expectedMenuItems)
+    {
+
+        List<String> actualValues = grid.getGridBar().getMenuText("Views");
+
+        return checker().withScreenshot(String.format("%s_Views_Menu_Error", screenShotPrefix))
+                .verifyEqualsSorted("Items under 'Views' menu not as expected.",
+                        expectedMenuItems, actualValues);
+
+    }
+
+    /**
+     * Helper to validate the columns in the grid.
+     *
+     * @param screenShotPrefix A string to put at the start of the screenshot name (if taken).
+     * @param grid A reference to the QueryGrid.
+     * @param expectedColumns A map of the expected columns. Key is the column label. The value indicates which icons, if any, should be present for the column.
+     * @return True if no errors, false otherwise.
+     */
+    private boolean validateGridColumns(String screenShotPrefix, QueryGrid grid, Map<String, Integer> expectedColumns)
+    {
+        checker().setErrorMark();
+
+        List<String> actualColumns = grid.getColumnNames();
+
+        checker().verifyEqualsSorted("Grid columns not as expected.",
+                expectedColumns.keySet(), actualColumns);
+
+        for(Map.Entry<String, Integer> entry : expectedColumns.entrySet())
+        {
+
+            if(actualColumns.contains(entry.getKey()))
+            {
+                if((entry.getValue() & 1) != 0)
+                {
+                    checker().verifyTrue(String.format("Column '%s' does not have the sorted icon.", entry.getKey()),
+                            grid.hasColumnSortIcon(entry.getKey()));
+                }
+
+                if((entry.getValue() & 2) != 0)
+                {
+                    checker().verifyTrue(String.format("Column '%s' does not have the filtered icon.", entry.getKey()),
+                            grid.hasColumnFilterIcon(entry.getKey()));
+                }
+
+            }
+
+        }
+
+        checker().screenShotIfNewError(String.format("%s_Views_Menu_Column_Error", screenShotPrefix));
+
+        return checker().errorsSinceMark() == 0;
```

Comment:
`screenshotIfNewError` resets the error mark. This will always return `true`.
I think it would make sense for `screenshotIfNewError` to return `true` if it took a screenshot.

Reply:
I've changed the DeferredErrorCollector.screenShotIfNewError to return a boolean. True if it took a screenshot, false otherwise.